units/event_metajob.go

```diff
@@ -0,0 +1,202 @@
+package units
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/notification"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/sometimes"
+	"github.com/pkg/errors"
+)
+
+const (
+	eventMetaJobName = "event-metajob"
+)
+
+func init() {
+	registry.AddJobType(eventMetaJobName, func() amboy.Job { return makeEventMetaJob() })
+}
+
+func notificationIsEnabled(flags *evergreen.ServiceFlags, n *notification.Notification) bool {
+	switch n.Subscriber.Type {
+	case event.GithubPullRequestSubscriberType:
+		return !flags.GithubStatusAPIDisabled
+
+	case event.JIRAIssueSubscriberType, event.JIRACommentSubscriberType:
+		return !flags.JIRANotificationsDisabled
+
+	case event.EvergreenWebhookSubscriberType:
+		return !flags.WebhookNotificationsDisabled
+
+	case event.EmailSubscriberType:
+		return !flags.EmailNotificationsDisabled
+
+	case event.SlackSubscriberType:
+		return !flags.SlackNotificationsDisabled
+
+	default:
+		grip.Alert(message.Fields{
+			"message": "notificationIsEnabled saw unknown subscriber type",
+			"cause":   "programmer error",
+		})
+	}
+
+	return false
+}
+
+type eventMetaJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	env      evergreen.Environment
+	events   []event.EventLogEntry
+	flags    *evergreen.ServiceFlags
+
+	Collection string `bson:"collection" json:"collection" yaml:"collection"`
+}
+
+func makeEventMetaJob() *eventMetaJob {
+	j := &eventMetaJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    eventMetaJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewEventMetaJob(collection string) amboy.Job {
+	j := makeEventMetaJob()
+	j.env = evergreen.GetEnvironment()
```

Comment:
should pass env in, or probably better yet, pass in a queue object.

Reply:
fixed